### ISSUE TITLE
Kube API Linter should be easy to extend

### ISSUE THREAD

**Issue by JoelSpeed** [kubernetes-sigs/kube-api-linter#80]
While the core of KAL should cover the majority of cases, there will always be cases where projects, companies, individuals would like to expand the set of rules that KAL implements.

To that end, I believe we should refactor KAL to allow such expansion. Likely in a way that allows folks to create their own plugin binary.

```
func New(pluginSettings any) ([]*analysis.Analyzer, error) {
	plugin, err := kubeapilinter.New(pluginSettings)
	if err != nil {
		return nil, fmt.Errorf("error creating plugin: %w", err)
	}

	plugin.Register(myanalyzer.Initializer, myotheranalyzer.Initializer)

	analyzers, err := plugin.BuildAnalyzers()
	if err != nil {
		return nil, fmt.Errorf("error building analyzers: %w", err)
	}

	return analyzers, nil
}
```

With an interface that allows users to add additional analyzers, and a library of helpers to help them to build those analyzers, folks should be able to extend the project.

The main open question I have for this is how would the configuration work.

Today, configuration and validation are in separate packages. We likely want to set a pattern for this and move the configuration and validation for each anaylzer into the package itself.

I could then imagine the configuration for custom rules looking something like:
```
kubeapilinter:
  lintersettings:
    custom:
    - name: mycustomrule
      config: 
        ...
```
Where the initializer for the custom plugin would return a name which would match with the name in the custom linters.

Some other potential things we could provide:
* Where we have common functions (e.g. A conflicts with B), allow those to be extended through configuration
* Where we have marker value type checking (e.g. A should be a number), allow configuration to extend the type checking

Do folks have other requirements or ideas they would like to see to make KAL easy to extend?

**Comment by JoelSpeed** [comment]
I'm going to find time to work on this, but if folks have ideas of how this should look, please let me know

**Comment by everettraven** [comment]
Could the config just generally look something like:
```yaml
kubeapilinter
  checks:
    - name: optionalorrequired
      config:
        ...
```

Would this simplify the process by not needing a separate layer to the configuration for "custom" linting rules?
